compiler/aarch64/codegen/OMRMemoryReference.cpp

```diff
@@ -1207,29 +1331,21 @@ uint8_t *OMR::ARM64::MemoryReference::generateBinaryEncoding(TR::Instruction *cu
                if (constantIsImm7(shifted))
                   {
                   *wcursor |= (shifted & 0x7f) << 15; /* imm7 */
-                  cursor += ARM64_INSTRUCTION_LENGTH;
                   }
                else
                   {
                   TR_ASSERT_FATAL(false, "Offset is too large for specified instruction.");
                   }
                }
-            else if (isExclusiveMemAccessInstruction(enc))
-               {
-               TR_ASSERT(displacement == 0, "Offset must be zero for specified instruction.");
```

Comment:
This assertion for ExclusiveMemAccessInstruction is lost.  The same for the assertion for AtomicOperationInstruction.

Reply:
I found the assertions in `estimateBinaryLength()`.